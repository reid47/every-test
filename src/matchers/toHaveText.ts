import { formatValue } from './common';

// declare global {
//   namespace jest {
//     interface Matchers<R> {
//       toHaveText: (expectedText: string) => any;
//     }
//   }
// }

export default function toHaveText(received, expectedText) {
  const actualText = received.textContent;
  const pass = actualText === expectedText;

  const message = pass
    ? `Expected element not to have text ${formatValue(expectedText)}.`
    : `Expected element to have text ${formatValue(
        expectedText
      )}, but actual text was ${formatValue(actualText)}.`;

  return { pass, message };
}
