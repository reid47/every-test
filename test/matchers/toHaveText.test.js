import toHaveText from '../../src/matchers/toHaveText';

describe('toHaveText', () => {
  test('passing', () => {
    const el = document.createElement('div');
    const text = document.createTextNode('hello');
    el.appendChild(text);

    const result = toHaveText(el, 'hello');
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('failing', () => {
    const el = document.createElement('div');
    const text = document.createTextNode('hello');
    el.appendChild(text);

    const result = toHaveText(el, 'something else');
    expect(result.pass).toBeFalsy();
    expect(result.message).toMatchSnapshot();
  });
});
