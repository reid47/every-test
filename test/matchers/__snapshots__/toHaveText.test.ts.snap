// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toHaveText failing 1`] = `"Expected element to have text \\"something else\\", but actual text was \\"hello\\"."`;

exports[`toHaveText passing 1`] = `"Expected element not to have text \\"hello\\"."`;
